From 93736ddb6f67e28b94cc4b69364d592e0e722086 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Wed, 27 Jan 2016 12:50:04 +0530
Subject: [PATCH 05/24] usbc/detect: set default host current in dfp mode.

The host current for pull-up should be set to USB_DEFAULT when
system settles at DFP.

Change-Id: Ib03043703d6a28a4c6e96aa9b5750a6c6c001284
Tracked-On:  https://jira01.devtools.intel.com/browse/OAM-18527
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/462780
---
 drivers/usb/typec/usb_typec_detect.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index efbdb57..8406b8e 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -522,6 +522,7 @@ static void detect_dfp_work(struct work_struct *work)
 	if (detect_src_attached(&phy->cc1, &phy->cc2)) {
 		use_cc = get_active_cc(phy->cc1.rd, phy->cc2.rd);
 		typec_setup_cc(phy, use_cc, TYPEC_STATE_ATTACHED_DFP);
+		typec_set_host_current(phy, TYPEC_CURRENT_USB);
 		/* enable VBUS */
 	} else if (detect_audio_attached(&phy->cc1, &phy->cc2)) {
 		dev_info(detect->phy->dev, "Audio Accessory Detected");
-- 
1.9.1

